publish-frontend/src/components/post-form.jsx

```diff
@@ -170,6 +141,63 @@ const PostForm = ({ tags, user, authors, post }) => {
     }
   };
 
+  // this ensures that the event listener has the latest version of handleSubmit.
+  // otherwise, it will have the initial version of the function, which will
+  // hold old values of state. This is because the event listener is created
+  // when the component is first rendered, and it will not be updated when
+  // the state changes.
+
+  const handleKeyDown = useCallback(
+    function (event) {
+      if ((event.ctrlKey || event.metaKey) && event.key === 's') {
+        event.preventDefault();
+        handleSubmit();
+      }
+    },
+    [handleSubmit]
+  );
```

Comment:
```suggestion
  function handleKeyDown(event) {
    if ((event.ctrlKey || event.metaKey) && event.key === 's') {
      event.preventDefault();
      handleSubmit();
    }
  }
```
This function is recreated on each render and a change to `handleSubmit` triggers a render, so it's automatically up-to-date.

Reply:
That's not the point of using the callback function. I know `handleSubmit`  it self wil re-render with the new values. But the version of `handleSubmit` in the `handleKeyDown` event listener will be holding an older state of `handleSubmit`  than what is stored on the server side. 

What will happen if we remove this:

```
add event listener to document -> 
(content in editor: <h1>hello</h1>) ->
 user types "world" ->
 user saves draft ->
 handleSubmit saves <h1>hello</h1>
```

Current behaviour:
```
add event listener to document -> 
(content in editor <h1>hello</h1>) -> 
user types "world" -> 
`useCallBack` tells to update event listener function -> 
user saves draft ->
old listener gets removed in `useEffect` ->
new listener gets added -> 
handleSubmit saves <h1>hello world</h1>
```

Reply:
`useCallback` is used to _prevent_ unnecessary updates to functions that do not need to change on every render. It can't be used to ensure that a function gets updated when it is not without `useCallback`.

Anyways, I tried the following steps locally:

1. Start with and empty post
2. type the word "hello"
3. pressed "Save as Draft"
4. reloaded the page (hello is still there)
5. selected "hello" and typed "world" over it
6. clicked "Save as Draft"
7. pressed `ctrl + s`
8. reloaded the page
9. saw "world"

Reply:
Yes everything should be working fine. But try to remove `useCallback` from the equation.